hls4ml/model/hls_layers.py

```diff
@@ -1122,6 +1122,181 @@ def initialize(self):
         else:
             self.set_attr('strategy', 'latency')
 
+class LSTM(Layer):
+    def initialize(self):
+        shape = [self.attributes['n_sequence_out'],int(self.attributes['recurr_n_out']/4)]
+        dims = ['N_SEQUENCE_OUT_{}'.format(self.index), 'N_LAYER_{}'.format(self.index)]
+        compression = self.model.config.get_compression(self)
+        reuse_factor = self.model.config.get_reuse_factor(self)
+        self.reuse_factor_recr = reuse_factor
+        if self.model.config.is_resource_strategy(self):
+            if self.model.config.backend.name == 'Vivado':
+                self.model.config.backend.set_closest_reuse_factor(self)
+                self.model.config.backend.set_closest_reuse_factor_recr(self)
```

Comment:
Since this function changes the object passed to it and doesn't return values that are assigned to an object, it would be better to extend `set_closest_reuse_factor` and not introduce the `_recr` variant and add `recr` parameter to it.